lib/trino-hive-formats/src/test/java/io/trino/hive/formats/ion/TestIonFormat.java

```diff
@@ -204,17 +204,53 @@ public void testDecimalPrecisionAndScale()
                 "{ amount: 1234.00, big_amount: 1234.00000 }"
                         + "{ amount: 1234d0, big_amount: 1234d0 }"
                         + "{ amount: 12d2, big_amount: 12d2 }"
-                        + "{ amount: 1234.000, big_amount: 1234.000000 }",
+                        + "{ amount: 1234.000, big_amount: 1234.000000 }"
+                        + "{ amount: 1234, big_amount: 1234 }", // these are both IonInts
                 List.of(new SqlDecimal(BigInteger.valueOf(123400), 10, 2), new SqlDecimal(BigInteger.valueOf(123400000), 25, 5)),
                 List.of(new SqlDecimal(BigInteger.valueOf(123400), 10, 2), new SqlDecimal(BigInteger.valueOf(123400000), 25, 5)),
                 List.of(new SqlDecimal(BigInteger.valueOf(120000), 10, 2), new SqlDecimal(BigInteger.valueOf(120000000), 25, 5)),
+                List.of(new SqlDecimal(BigInteger.valueOf(123400), 10, 2), new SqlDecimal(BigInteger.valueOf(123400000), 25, 5)),
                 List.of(new SqlDecimal(BigInteger.valueOf(123400), 10, 2), new SqlDecimal(BigInteger.valueOf(123400000), 25, 5)));
     }
 
     @Test
-    public void testOversizeOrOverpreciseDecimals()
+    public void testNumbersTooBigForShortDecimal()
     {
-        // todo: implement
+        RowType rowType = RowType.rowType(
+                field("amount", DecimalType.createDecimalType(4, 2)));
+
+        List<String> ions = List.of(
+                "{ amount: 123.4 }",
+                "{ amount: 1.234 }",
+                "{ amount: 123 }",
+                "{ amount: 1234d-10 }",
+                "{ amount: 1234d2 }");
+
+        for (String ionText : ions) {
+            Assertions.assertThrows(
+                    NumberFormatException.class,
+                    () -> assertValues(rowType, ionText, List.of()));
+        }
+    }
+
+    @Test
+    public void testNumbersTooBigForDecimal128()
+    {
+        RowType rowType = RowType.rowType(
+                field("amount", DecimalType.createDecimalType(20, 2)));
+
+        List<String> ions = List.of(
+                "{ amount: 12345678901234567890.4 }",
+                "{ amount: 1.234 }",
```

Comment:
Nit: Would it be more explicit to modify the test case from `1.234` to `12345678901234567890.123?` My understanding is that we're testing numbers too large for Decimal128, specifically where the scale overflows. To make this scenario more explicit, should we ensure that the test case itself is a Decimal128 with an overflow scale?

Reply:
That is a different than this but yea that would be a good one to prove.